charts/kube-prometheus-stack/charts/crds/crds/crd-prometheuses.yaml

```diff
@@ -68,6 +68,9 @@ spec:
             description: 'Specification of the desired behavior of the Prometheus
               cluster. More info: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status'
             properties:
+              automountServiceAccountToken:
+                description: Controls whether a service account token is automatically mounted for the Prometheus pods.
+                type: boolean
```

Comment:
Could you please explain, how this is working?

Reply:
> Could you please explain, how this is working?

As said in the issue #4148 under the solution section that  `crd-prometheuses.yaml` does not have the argument option automountServiceAccountToken available, in that case the possibility to set `automountServiceAccountToken: False` is not possible as of now, so I defined `automountServiceAccountToken` as custom resource definition to prevent auto mount of service account token.

If I missed out on something and more changes are needed please let me know I would be happy to contribute and learn.

Reply:
But if you define a new argument, who will take note of it? Without a technical implementation, the parameter does not have any effect.

Reply:
> But if you define a new argument, who will take note of it? Without a technical implementation, the parameter does not have any effect.

yes you're right, can you please share any reference that might help me to learn and look into this.

Reply:
> But if you define a new argument, who will take note of it? Without a technical implementation, the parameter does not have any effect.


> But if you define a new argument, who will take note of it? Without a technical implementation, the parameter does not have any effect.
> 
> Looking at the source code of the prometheus-operator
> 
> https://github.com/prometheus-operator/prometheus-operator/blob/18265de80ea60ab1ab31b16c5c6357324d9fe483/pkg/prometheus/server/statefulset.go#L475
> 
> The operator always set `automountServiceAccountToken` to `true`. Define a new argument on the CRDs will not change the behavior here.

so if I set it to false would it be right ?

Reply:
> > But if you define a new argument, who will take note of it? Without a technical implementation, the parameter does not have any effect.
> 
> yes you're right, can you please share any reference that might help me to learn and look into this.

Some out of the context here. Here we just provide the manifests for the deployment. For technical implementation, for you to write an issue here: https://github.com/prometheus-operator/prometheus-operator

Once its getting accepted and implemented, it will be included here. Unless then, I will close this.

Reply:
@jkroepke as https://github.com/prometheus-operator/prometheus-operator/issues/6270 is closed is this PR relevant now ??

Reply:
Yes I was also looking forward to changing it to true, but I was wondering why does the original issue #4148 wants to set `automountServiceAccountToken: False` as it will introduce a breaking change.

Reply:
The general recommendation is `automountServiceAccountToken=false`. However that case, Prometheus would stop working. The sidecar `config-reloader` requires Kubernetes API access to fetch the Prometheus which is stored as secret.

Additionally, Prometheus itself requires API access too. `spec.automountServiceAccountToken=false` will work, if you are not using any CRDs like ServiceMonitor, Probe or PrometheusRules.

#4148 goes a bit crazy by disable the Kubernetes autoMount and want to manually mount service Accounts. 

This pattern is mandatory if you have silly gov rules which denies autoMount but does not care, if you just manually mount the token.

Reply:
Thank you so much for this great explanation.

Reply:
@jkroepke is the new version released yet ?

Reply:
Yes, 0.74.0 is included in kube-prometheus-stack, feel free to setup a new PR

Reply:
If you could reopen this, I'll update the branch, set `automountServiceAccountToken=false` and push the latest changes.

Reply:
Please leave the default value (true) thanks

Reply:
Yeah, sorry I wrote false by mistake 🤣

Reply:
Hey there @jkroepke I just went through the latest code the CRD has already been defined, but the it not yet included in `charts/kube-prometheus-stack/values.yaml` and `charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml`,

Reply:
could you please help me out I'm getting confused seems like it has been implemented already :

https://github.com/prometheus-community/helm-charts/blob/8b9a537b5a2b743063fc4bf1f1e53341aafe7706/charts/kube-prometheus-stack/charts/crds/crds/crd-prometheuses.yaml#L1948-L1956

https://github.com/prometheus-community/helm-charts/blob/8b9a537b5a2b743063fc4bf1f1e53341aafe7706/charts/kube-prometheus-stack/values.yaml#L2456

Reply:
Of couse, the CRD was updated at upstream. 

The line you mention belongs to https://github.com/prometheus-community/helm-charts/blob/8b9a537b5a2b743063fc4bf1f1e53341aafe7706/charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/deployment/deployment.yaml#L126, but not the Prometheus CR.

Reply:
Seems like I got confused, here's the one which had me in doubt: https://github.com/prometheus-community/helm-charts/blob/8b9a537b5a2b743063fc4bf1f1e53341aafe7706/charts/kube-prometheus-stack/values.yaml#L2944

Reply:
that the properties of Prometheus SA , your initial PR covers the setting at the Deployment/Statefuleset.

Reply:
> that the properties of Prometheus SA , your initial PR covers the setting at the Deployment/Statefuleset.

yeah actually it's been a while since I opened this PR, so it took me a bit

Reply:
Well @jkroepke I've pushed the latest changes, do check it out, however while I was trying to test in my local system, it throwed error on the files I didn't even touch, seems like the bug was already in the upstream.

```
$ helm install  my-kube kube-prometheus-stack-60.4.0.tgz
Error: INSTALLATION FAILED: YAML parse error on kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/job-patch/job-createSecret.yaml: error converting YAML to JSON: yaml: line 45: did not find expected key
```

I checked the file for any potential bugs, but it sorta seems like indentation bug, but still I'm not very much inclined towards fixing it in this PR itself, I am thinking of fixing it in a separate PR.

Reply:
please create a dedicated issue, thanks!